src/components/templates/SingleReferenceReadonly/index.ts

```diff
@@ -0,0 +1,95 @@
+import { html, customElement, property } from '@lion/core';
+import { BridgeBase } from '../../../bridge/BridgeBase';
+// NOTE: you need to import ANY component you may render.
+
+// import the component's styles as HTML with <style>
+import '../../../components/forms/SemanticLink';
+
+
+// Declare that PCore will be defined when this code is run
+
+
+@customElement('single-reference-readonly')
+class SingleReferenceReadonly extends BridgeBase {
+  @property( {attribute: false, type: Object } ) pConn; 
+  constructor() {
+    //  Note: BridgeBase constructor has 2 optional args:
+    //  1st: inDebug - sets this.bLogging: false if not provided
+    //  2nd: inLogging - sets this.bLogging: false if not provided.
+    //  To get started, we set both to true here. Set to false if you don't need debugger or logging, respectively.
+    super(false, false);
+    if (this.bLogging) { console.log(`${this.theComponentName}: constructor`); }
+    if (this.bDebug){ debugger; }
+
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    if (this.bLogging) { console.log(`${this.theComponentName}: connectedCallback`); }
+    if (this.bDebug){ debugger; }
+
+    //NOTE: Need to bind the callback to 'this' so it has this element's context when it's called.
+    this.registerAndSubscribeComponent(this.onStateChange.bind(this));
+    
+  }
+
+
+  disconnectedCallback() {
+    // The super call will call storeUnsubscribe...
+    super.disconnectedCallback();
+    if (this.bLogging) { console.log(`${this.theComponentName}: disconnectedCallback`); }
+    if (this.bDebug){ debugger; }
+
+  }
+  
+  /**
+   * updateSelf
+   */
+  updateSelf() {
+    if (this.bLogging) { console.log(`${this.theComponentName}: updateSelf`); }
+    if (this.bDebug){ debugger; }
+
+  }
+
+  /**
+   * The `onStateChange()` method will be called when the state is updated.
+   *  Override this method in each class that extends BridgeBase.
+   *  This implementation can be used for common code that should be done for
+   *  all components that are derived from BridgeBase
+   */
+  onStateChange() {
+    if (this.bLogging) { console.log(`${this.theComponentName}: onStateChange`); }
+    if (this.bDebug){ debugger; }
+
+    const bShouldUpdate = super.shouldComponentUpdate();
+
+    if (bShouldUpdate) {
+      this.updateSelf();
+    }
+  }
+
+  getSingleReferenceHtml(): any {
+    const singleRefHtml = html`<semantic-link .pConn=${this.pConn}></semantic-link>`;
+
+    return singleRefHtml;
+  }
+
+  render(){
+    console.log('pConnn singleref', this.pConn);
```

Comment:
pConnn has 3 ds in this console.log but I think this should be removed or commented out to remove console clutter.

Reply:
removed the console and updated the PR